src/org/omegat/core/CoreEvents.java

```diff
@@ -123,6 +126,7 @@ public static void fireProjectChange(final IProjectEventListener.PROJECT_CHANGE_
                     log("ERROR_EVENT_PROJECT_CHANGE", t);
                 }
             }
+            isInProjectEventLoop = false;
```

Comment:
In order to access to the status of listener, you can use `PropertyChangeListener` that is Java standard mechanism and many swing components implemented.